frontend/src/components/collective/CollectiveExpenseItem.js

```diff
@@ -39,6 +39,13 @@ const CollectiveExpenseItem = ({
 
   let src, srcSet, href, receiptMessage, receiptHelpTip;
 
+  let submittedByName = user && user.name;
+
+  // show email only if the user can view receipt
+  if (canViewReceipt) {
```

Comment:
I agree. I was just worried that in the future if we allow visitors to cover an expense, they may have the right to canViewReceipt, which would make that email address public. I want to avoid that mistake.